### PR TITLE
Implement Edit Task prompt

### PR DESCRIPTION
**Pull request by kavisha-perera**
## Purpose

The purpose of this PR is to fix part of https://github.com/we-huddle/server/issues/19 

## Goals

Created the Dummy Edit Task UI component 

## Screenshots

image

## Related PRs
 
https://github.com/we-huddle/client/pull/15

### CODE REVIEW

File: src/scenes/TaskDetailsView/components/EditTaskPrompt/index.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+              </Button>
+            </div>
+          </form>
+        </div>
+      </Modal.Body>
+    </Modal>
+  );
+}
+
+export default EditTaskPrompt;
```
Comment (Gravewalker666) [inline src/scenes/TaskDetailsView/components/EditTaskPrompt/index.tsx:80]: Put a new line at the end of the file (EOF)

File: src/scenes/TaskDetailsView/components/EditTaskPrompt/index.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import {Button, Label, Modal, Textarea, TextInput} from "flowbite-react";
+
+interface EditTaskPromptProps {
+  show: boolean,
+  onClose: () => void,
+}
+
+function EditTaskPrompt({ show, onClose }: EditTaskPromptProps) {
```
Comment (Gravewalker666) [inline src/scenes/TaskDetailsView/components/EditTaskPrompt/index.tsx]: Let's take the task as a property and populate the form with those data

File: src/scenes/TaskDetailsView/index.tsx [hunk 3]
```diff
@@ -55,17 +67,19 @@ function TaskDetailsView({ isAgentView }: TaskDetailsViewProps){
...
+                  <Button onClick={() => setIsEditModalVisible(true)}>
+                      Edit
+                      <HiPencil className="ml-2" />
+                  </Button>
+                  <Button color="failure">
+                      Delete
+                      <HiTrash className="ml-2" />
+                  </Button></>
+              )}
+
```
Comment (Gravewalker666) [inline src/scenes/TaskDetailsView/index.tsx]:
```suggestion
              {isAgentView && profile?.role === Profile.Role.HuddleAgent && (
                  <>
                    <Button onClick={() => setIsEditModalVisible(true)}>
                        Edit
                        <HiPencil className="ml-2" />
                    </Button>
                    <Button color="failure">
                        Delete
                        <HiTrash className="ml-2" />
                    </Button>
                  </>
              )}
```
Not that important but better to have them properly formatted

**Review by Gravewalker666 — CHANGES_REQUESTED** [review]
Requested some changes

File: src/scenes/TaskDetailsView/components/EditTaskPrompt/index.tsx (hunk 1 above)
Comment (kavisha-perera) [reply src/scenes/TaskDetailsView/components/EditTaskPrompt/index.tsx:80]: fixed

File: src/scenes/TaskDetailsView/index.tsx (hunk 3 above)
Comment (kavisha-perera) [reply src/scenes/TaskDetailsView/index.tsx]: fixed

File: src/scenes/TaskDetailsView/components/EditTaskPrompt/index.tsx (hunk 2 above)
Comment (Gravewalker666) [reply src/scenes/TaskDetailsView/components/EditTaskPrompt/index.tsx]: This is a good working solution, well done.
We can further improve performance if we can somehow change this implementation from making another network call to getting the task passed down to the component via `props`. Let's edit the edit task prompt props to include the task in it. And then take that input parameter and render the form. 
Then you'll have to pass down the task object to this component from task details view.

**Review by Gravewalker666 — CHANGES_REQUESTED** [review]

**Review by Gravewalker666 — APPROVED** [review]
Great work @kavisha-perera thanks for getting this done

**Pull request merged**